### PR TITLE
chore!: drop pydantic 1 support

### PR DESCRIPTION
**Pull request by eddyxu**
drop pydantic version 1 support

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -195,31 +195,3 @@ jobs:
       # Make sure wheels are not included in the Rust cache
       - name: Delete wheels
         run: rm -rf target/wheels
-  pydantic1x:
```
Comment (wjones127) [inline .github/workflows/python.yml:198]: Could you keep a version of this job around called 'min-deps' or something? we also use this to test the minimum version of `pyarrow`, for example.

Comment (eddyxu) [reply .github/workflows/python.yml:198]: will do

**Comment by wjones127** [comment]
Closing this for now. We can re-open this later if we want.
